packages/x-charts/src/models/seriesType/config.ts

```diff
@@ -7,35 +7,68 @@ import { DefaultizedProps, MakeOptional } from '../helpers';
 import { StackingGroupsType } from '../../internals/stackSeries';
 import { SeriesId } from './common';
 
-interface ChartsSeriesConfig {
+export interface ChartsSeriesConfig {
   bar: {
+    /**
+     * Series type when passed to the formatter (some ids are defaultised to simplify the DX)
+     */
     seriesInput: DefaultizedProps<BarSeriesType, 'id'> & { color: string };
+    /**
+     * Series type when stored in the context (with all the preprocessing added))
+     */
     series: DefaultizedBarSeriesType;
-    canBeStacked: true;
+    /**
+     * Series typing such that the one user need to provide
+     */
+    seriesProp: BarSeriesType;
     itemIdentifier: BarItemIdentifier;
+    canBeStacked: true;
+    cartesian: true;
   };
   line: {
     seriesInput: DefaultizedProps<LineSeriesType, 'id'> & { color: string };
     series: DefaultizedLineSeriesType;
-    canBeStacked: true;
+    seriesProp: LineSeriesType;
     itemIdentifier: LineItemIdentifier;
+    canBeStacked: true;
+    cartesian: true;
   };
   scatter: {
     seriesInput: DefaultizedProps<ScatterSeriesType, 'id'> & { color: string };
     series: DefaultizedScatterSeriesType;
+    seriesProp: ScatterSeriesType;
     itemIdentifier: ScatterItemIdentifier;
+    cartesian: true;
   };
   pie: {
     seriesInput: Omit<DefaultizedProps<PieSeriesType, 'id'>, 'data'> & {
       data: (MakeOptional<PieValueType, 'id'> & { color: string })[];
     };
     series: DefaultizedPieSeriesType;
+    seriesProp: PieSeriesType<MakeOptional<PieValueType, 'id'>>;
     itemIdentifier: PieItemIdentifier;
   };
 }
 
-export type CartesianChartSeriesType = 'bar' | 'line' | 'scatter';
-export type ChartSeriesType = 'bar' | 'line' | 'scatter' | 'pie';
+export type ChartSeriesType = keyof ChartsSeriesConfig;
+
+export type CartesianChartSeriesType = keyof Pick<
+  ChartsSeriesConfig,
+  {
+    [Key in ChartSeriesType]: ChartsSeriesConfig[Key] extends { cartesian: true } ? Key : never;
+  }[ChartSeriesType]
+>;
+
+export function isCartesianSeriesType(seriesType: string): seriesType is CartesianChartSeriesType {
+  return ['bar', 'line', 'scatter', 'heatmap'].includes(seriesType);
```

Comment:
Should we have a singleton where we can register different types of cartesian series? something like

```tsx
class CartesianSeriesTypes {
  _types: Set<ChartSeriesType> = new Set(['bar', 'line', 'scatter'])

  constructor() {
    if (CartesianSeriesTypes._instance) {
      return CartesianSeriesTypes._instance
    }
    CartesianSeriesTypes._instance = this;
  }

  addType(value: 'string') { _types.add(value) }
  getTypes() { return _types }
}
```

this would allow us to register new types from other packages. We could also make the class a bit more generic and instead allow registering full configurations, eg: `SeriesTypes.register({ name: "heatmap", formatter: (v) => v.foo, isCartesian: true })`

Reply:
Then you would instantiate it in the index.ts of each package.

Can be an elegant solution for the `isCartesianSeriesType`. I'm not fully in favor of expanding this strategy to other aspect, because one advantage of plugins is the tree shaking. Since the heatmap logic is only imported in tht `src/Heatmap/` folder, if you don't import it you will not bundle its logic. Same for the tree view, radar, and other upcoming charts. Not sure it will be the same with this approach

Reply:
Technically we could only register if the user imports `charts/heatmap` or something like that. 

I've been thinking about the plugin, and the drawback might be that we will need to pass the configuration around to everything that would need it 😅

Reply:
I did the modification for the config. I did not used it in the pro package yet, because otherwise TS will complain that `'heatmap'` is not a series type yet. Will be added in the next PR